### PR TITLE
Lint all the things

### PR DESCRIPTION
**Pull request by sandersky**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

screen shot 2017-02-06 at 11 26 33 pm

# CHANGELOG

* **Added** new Node executables for linting: `lint-all-the-things`, `lint-docker`, `lint-htmlbars` (previously `template-lint`), `lint-javascript`, `lint-markdown`, and `lint-sass`.

### CODE REVIEW

File: .sass-lint.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    "bem-depth": 0,
+    "border-zero": 2,
+    "brace-style": 2,
+    "clean-import-paths": 2,
+    "empty-args": 2,
+    "hex-length": 2,
+    "hex-notation": 2,
+    "indentation": 2,
+    "leading-zero": 2,
+    "nesting-depth": 0,
```
Comment (job13er) [inline .sass-lint.json]: I thought by default we wanted a nesting depth defined. We've had to override in a few apps to get them to work until they can be refactored, but we probably shouldn't default to not checking it.

File: cli/lint-sass.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * List of glob directories for where to find SASS files
+ * @type {Array.<string>}
+ */
+const TEMPLATE_FILE_LOCATIONS = [
+  'addon/**/*.sass',
+  'addon/**/*.scss',
+  'app/**/*.sass',
+  'app/**/*.scss',
+  'tests/**/*.sass',
```
Comment (job13er) [inline cli/lint-sass.js:26]: Do we have an `.sass` files or care about linting them? I thought we agreed on `.scss`?

File: .sass-lint.json (hunk 1 above)
Comment (sandersky) [reply .sass-lint.json]: Ah, I copied the rules from one of our apps, I guess this was changed there. Good catch.

File: cli/lint-sass.js (hunk 2 above)
Comment (sandersky) [reply cli/lint-sass.js:26]: I believe we did agree on `.scss` but if someone tries to use `.sass` I see no harm in linting it too.

**Comment by sandersky** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ciena-blueplanet/ember-test-utils/pull-request/29/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-blueplanet/ember-test-utils)

**Pull request merged**